metadata-service/auth-impl/src/main/java/com/datahub/authentication/token/StatelessTokenService.java

```diff
@@ -95,20 +98,23 @@ public String generateAccessToken(
    * Note that the caller of this method is expected to authorize the action of generating a token.
    */
   @Nonnull
-  public String generateAccessToken(@Nonnull final String sub, @Nonnull final Map<String, Object> claims, final long expiresInMs) {
+  public String generateAccessToken(@Nonnull final String sub, @Nonnull final Map<String, Object> claims, final Optional<Long> expiresInMs) {
     Objects.requireNonNull(sub);
     Objects.requireNonNull(claims);
     final JwtBuilder builder = Jwts.builder()
       .addClaims(claims)
-      .setExpiration(new Date(System.currentTimeMillis() + expiresInMs))
       .setId(UUID.randomUUID().toString())
       .setSubject(sub);
+
+    if (expiresInMs.isPresent()) {
+      builder.setExpiration(new Date(System.currentTimeMillis() + expiresInMs.get()));
```

Comment:
Question: 

If we don't provide an expiration, the signing of the JWT will still work?

Reply:
Please refer [this test-case](https://github.com/datahub-project/datahub/pull/5433/files#diff-68a1c8e0196ee2b55feef1a2739ae27203c3331c955a5cfa25daad1d97f7bf11R68).

Does this address the question?

(Also, I have manually tested that a token created with NO_EXPIRY during the graphql API can be used for ingestion using CLI.)

Can you please elaborate if any further cases are to be tested?

Reply:
Yes thank you - as long as you've tested it